src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -111,52 +111,65 @@ export default class AircraftController {
      * This is the entry method for creating new departing and arriving aircraft.
      * This method should only be called as a callback from a `SpawnScheduler` timer.
      *
-     * @method createAircraftWithSpawnModel
+     * @method createAircraftWithSpawnPatternModel
      * @param spawnPatternModel {SpawnPatternModel}
+     * @private
      */
-    createAircraftWithSpawnModel = (spawnPatternModel) => {
+    createAircraftWithSpawnPatternModel = (spawnPatternModel) => {
         const initializationProps = this._buildAircraftProps(spawnPatternModel);
-        const aircraftModel = new AircraftInstanceModel(initializationProps);
 
-        this.addItem(aircraftModel);
+        this._createAircraftWithInitializationProps(initializationProps);
     }
 
     /**
-     * @for AircraftController
-     * @method aircraft_auto_toggle
+     * Build aircraft props for `spawnPatternModel` and add
+     * preSpawn data to `baseAircraftProps`.
+     *
+     * Used when creating aircraft already along an arrival route on load
+     * or on airport change.
+     *
+     * This method should be called direectly and not via a timer callback
+     *
+     * This hooks into the same method used to build spawnning aircraft
+     * and simply adds another layer on top of that to build a preSpawn aircraft.
+     *
+     * @method createPreSpawnAircraftWithSpawnPatternModel
+     * @param  spawnPatternModel {SpawnPatternModel}
+     * @private
      */
-    aircraft_auto_toggle() {
-        this.aircraft.auto.enabled = !this.aircraft.auto.enabled;
-    }
+    createPreSpawnAircraftWithSpawnPatternModel = (spawnPatternModel) => {
+        for (let i = 0; i < spawnPatternModel.preSpawnAircraftList.length; i++) {
+            const preSpawnHeadingAndPosition = spawnPatternModel.preSpawnAircraftList[i];
+            const baseAircraftProps = this._buildAircraftProps(spawnPatternModel);
+            const initializationProps = Object.assign({}, baseAircraftProps, preSpawnHeadingAndPosition);
+
+            this._createAircraftWithInitializationProps(initializationProps);
+        }
+    };
 
     /**
+     * Accept a pre-built object that can be used to create an `AircraftInstanceModel`
+     * and then add it to the collection.
+     *
+     * This could be a spawnning aircraft or one that already exists along a route.
+     *
      * @for AircraftController
-     * @method isCallsignInList
-     * @param callsign {string}
-     * return {boolean}
+     * @method _createAircraftWithInitializationProps
+     * @param initializationProps {object}
+     * @private
      */
-    isCallsignInList(callsign) {
-        // TODO: use a getter in airlineCollection that looks at each `AirlineModel.flightNumbers`
-        return this.aircraft.callsigns.indexOf(callsign) !== -1;
+    _createAircraftWithInitializationProps(initializationProps) {
```

Comment:
Good question. add is certainly part of it, but that is really a side effect.  The meat of this method (all three lines of it) is instantiating, or _creating_, an `AircraftInstanceModel`.